New package: jamulus-3.6.2

kawaiiamber opened the pull request:
Closes [this](https://github.com/void-linux/void-packages/issues/27738), `./xbps-src pkg jamulus` passes locally on my machine, don't know how cross-platform will do.

abenson commented on srcpkgs/jamulus/template:
```diff
@@ -0,0 +1,16 @@
+pkgname=jamulus
+version=3.6.2
+revision=1
+archs="x86_64* i686"
```
Why restrict the archs?

kawaiiamber replied:
The CI fails on other archs otherwise. I could get rid of this to display the errors if you'd like.

abenson replied:
Fix the cross-build.

abenson replied:
Also, if its not possible to cross-build, use `nocross`, don't restrict the arches.

kawaiiamber replied:
`nocross="Compilation fails on some archs"`?

kawaiiamber replied:
Set `nocross` after `checksum`

abenson commented on srcpkgs/jamulus/template:
```diff
@@ -0,0 +1,16 @@
+build_style=qmake
+configure_args="Jamulus.pro"
+hostmakedepends="pkg-config"
+makedepends="qt5-declarative-devel jack-devel qjackctl"
```
`qjackctl` is makedepends?

kawaiiamber replied:
It was listed as part of the build process [here](https://jamulus.io/wiki/Installation-for-Linux), I could try moving it to `depends` instead if that's more appropraite.

kawaiiamber replied:
I see what you mean. After reading through the jamulus documentation more, a user would just install the `qjackctl` package on their own if that's how they wanted to set up jack with it. Removed `qjackctl` from `makedepends`.

abenson commented on srcpkgs/jamulus/template:
```diff
@@ -0,0 +1,16 @@
+maintainer="KawaiiAmber <japaneselearning101@gmail.com>"
+license="GPL-2.0-only"
+homepage="https://jamulus.io/"
+distfiles="https://phoenixnap.dl.sourceforge.net/project/llcon/Jamulus/${version}/Jamulus-${version}.tar.gz"
```
`$SOURCEFORGE_SITE`

kawaiiamber replied:
`distfiles="${SOURCEFORGE_SITE}/project/llcon/Jamulus/${version}/Jamulus-${version}.tar.gz"`?

abenson commented on srcpkgs/jamulus/template:
```diff
@@ -0,0 +1,16 @@
+makedepends="qt5-declarative-devel jack-devel qjackctl"
+short_desc="Play music online. With friends. For free"
+maintainer="KawaiiAmber <japaneselearning101@gmail.com>"
+license="GPL-2.0-only"
```
This is `GPL-2.0-or-later`

abenson commented:
Try and get cross-build working. 

    => ERROR: jamulus-3.6.2_1: Could not find qmake - missing in hostdepends?

Might be as simple as adding `qt5-devel` to `hostmakedepends`

kawaiiamber commented:
Added `qt5-devel` to `hostmakedepends`

kawaiiamber commented:
On armv6l-musl:
```
ERROR: [trans] failed to download `xcb-util-wm-0.4.1_3' package from `http://alpha.us.repo.voidlinux.org/current/musl': Transient resolver failure
Transaction failed! see above for errors.
=> ERROR: Please see above for the real error, exiting...
Error: Process completed with exit code 1.
```
I'm not sure why it's faling to download the `xcb-util-wm-0.4.1_3` package.

abenson commented:
That's just the CI being wonky.

kawaiiamber commented:
Oh, is there a way to have CI check again?

kawaiiamber commented:
```
Run # Sync and upgrade once, assume error comes from xbps update
ERROR: [reposync] failed to fetch file `https://alpha.de.repo.voidlinux.org/current/x86_64-repodata': Transient resolver failure
[*] Updating repository `https://alpha.de.repo.voidlinux.org/current/x86_64-repodata' ...
Package 'xbps' not found in repository pool.
Error: Process completed with exit code 2.
```
??????? Something down with CI? It was passing all checks before I added `jack` to `depends`. Now, with or without it, it fails on the setup stage.

ericonr commented on srcpkgs/jamulus/template:
```diff
@@ -0,0 +1,20 @@
+maintainer="KawaiiAmber <japaneselearning101@gmail.com>"
+license="GPL-2.0-or-later"
+homepage="https://jamulus.io/"
+distfiles="${SOURCEFORGE_SITE}/project/llcon/Jamulus/${version}/Jamulus-${version}.tar.gz"
```
Why sourceforge and not their github repo? It's recommended in https://jamulus.io/wiki/Installation-for-Linux

kawaiiamber replied:
Weird. When I went to get the distfiles, I just went to jamulus.io and clicked "download now," which led to their sourceforge. I didn't even notice that they recommend the github in their guide. I'll change it to use github when I get back home.

kawaiiamber replied:
Updated to use github distfiles.

ericonr commented on srcpkgs/jamulus/template:
```diff
@@ -0,0 +1,20 @@
+distfiles="${SOURCEFORGE_SITE}/project/llcon/Jamulus/${version}/Jamulus-${version}.tar.gz"
+checksum=86d38f7d22b49720a79a246918cb8f16a690bbc7d6cbc8707e93694bab1a074c
+
+if [ "${CROSS_BUILD}" ]; then
+	hostmakedepends+=" qt5-devel"
```
Feel free to put this in `hostmakedepends` unconditionally.

kawaiiamber replied:
For the sake of maintaining a cross platform package, I like knowing what is needed specifically for cross builds. I'd prefer o keep it this way for the sake of maintaining, but if there's a general preference in packages not to do this, let me know and I can add `qt5-devel` unconditionally to `hostmakedepends` when I get home.

ericonr replied:
It's ok to leave as is if you prefer it.

ericonr reviewed (commented):
@38github have you managed to test the resulting package?

kawaiiamber commented:
> @38github have you managed to test the resulting package?

I tested it myself a while ago. `xbps-install --repository=hostdir/binpkgs/jamulus jamulus`. I myself don't have instruments that can connect to jack / the computer, but was able to browse and connect to public instances and listen to random groups play music. @38github , if you have instruments that can play on linux, I'd love if you could test this package as I can't.

38github commented:
After a lot of tinkering I managed to setup it all up for building and managed to use one of the templates posted somewhere here and it built and works! Audio from the sound card is coming into Jamulus and sound is coming out all using JACK.

Thank you all! Now I have a complete package I can make for friends that wants to use Jamulus: EFI installation to USB (needed to add some files afterwards using chrrot), XFCE, Firefox, Jamulus, Qjackctl and add user to audio group. Great!

kawaiiamber commented:
> After a lot of tinkering I managed to setup it all up for building and managed to use one of the templates posted somewhere here and it built and works! Audio from the sound card is coming into Jamulus and sound is coming out all using JACK.
> 
> Thank you all! Now I have a complete package I can make for friends that wants to use Jamulus: EFI installation to USB (needed to add some files afterwards using chrrot), XFCE, Firefox, Jamulus, Qjackctl and add user to audio group. Great!

Nice, thanks for testing it. I hope that "one of the templates posted somewhere here" is referring to this one. :) Only thing that left that I will do it change `distfiles` to use github instead of sourceforge when I get home, but this shouldn't affect the condition of the template.

38github commented:
That's amazing to see it compile here in github and being able to see the progress. Wow.

kawaiiamber commented:
Okay, I updated to use distfiles from github rather than sourceforge. It should perform the same.

ericonr commented on srcpkgs/jamulus/template:
```diff
@@ -0,0 +1,22 @@
+maintainer="KawaiiAmber <japaneselearning101@gmail.com>"
+license="GPL-2.0-or-later"
+homepage="https://jamulus.io"
+_github="https://github.com/corrados/${pkgname}"
+distfiles="${_github}/archive/${_version}.tar.gz"
```
```suggestion

distfiles="https://github.com/corrados/jamulus/archive/${_version}.tar.gz"
```

kawaiiamber replied:
Fair enough. Changed

ericonr commented on srcpkgs/jamulus/template:
```diff
@@ -0,0 +1,22 @@
+pkgname=jamulus
+version=3.6.2
+revision=1
+_version=r$(echo ${version} | sed 's/\./_/g')
```
```suggestion
_version=r${version//./_}
```

kawaiiamber replied:
This... kind of blew my mind. Thanks for the tip! :)

ericonr commented on srcpkgs/jamulus/template:
```diff
@@ -0,0 +1,22 @@
+configure_args="Jamulus.pro"
+hostmakedepends="pkg-config"
+makedepends="qt5-declarative-devel jack-devel"
+depends="jack"
```
I don't think this should be here; a jack server can be provided by some alternative application, like pipewire.

kawaiiamber replied:
I didn't know about things like pipewire - thanks for the tip! :) removed `depends`.

ericonr commented:
Hm, could you try decreasing the version locally and testing if `./xbps-src update-check jamulus` works? There's a chance it won't detect their weird version name, so you'd need to add an `update` file for it.

kawaiiamber commented:
> Hm, could you try decreasing the version locally and testing if `./xbps-src update-check jamulus` works? There's a chance it won't detect their weird version name, so you'd need to add an `update` file for it.

Okay, I cleaned the masterdir with `./xbps-src clean` and replaced the template with the previous version:
```
# Template file for 'jamulus'
pkgname=jamulus
version=3.6.1
revision=1
_version=r${version//./_}
wrksrc=${pkgname}-${_version}
build_style=qmake
configure_args="Jamulus.pro"
hostmakedepends="pkg-config"
makedepends="qt5-declarative-devel jack-devel"
short_desc="Play music online. With friends. For free"
maintainer="KawaiiAmber <japaneselearning101@gmail.com>"
license="GPL-2.0-or-later"
homepage="https://jamulus.io"
distfiles="https://github.com/corrados/jamulus/archive/${_version}.tar.gz"
checksum=24e9d7c69fcf949261fef39160b716375ac24afd2ba948d44d013a8b2ed1287c

if [ "${CROSS_BUILD}" ]; then
	hostmakedepends+=" qt5-devel"
fi
```
I then built the package `./xbps-src pkg jamulus`, then checked for an update:
```
[amber@otaku ~/projx/repos/void-packages]$ ./xbps-src update-check jamulus
NO VERSION found for jamulus
```
It would appear as though you're right. I don't know anything about `update` files.

ericonr commented:
@kawaiiamber you can add a `srcpkgs/jamulus/update` file with the following contents:

```
pattern="r\K[0-9_]+_[0-9_]+"
```

kawaiiamber commented:
> @kawaiiamber you can add a `srcpkgs/jamulus/update` file with the following contents:
> 
> ```
> pattern="r\K[0-9_]+_[0-9_]+"
> ```

Okay, added

ericonr commented:
Almost went deaf, but that's on me.
